velox/exec/ContainerRowSerde.cpp

```diff
@@ -332,6 +332,28 @@ int compareSwitch(
     vector_size_t index,
     CompareFlags flags);
 
+template <typename T>
+int32_t compareNative(
```

Comment:
Should we move the tests to velox/exec/tests/ContainerRowSerdeTest.cpp ?

It would be nice to update the comments for the ContainerRowSerde API to clarify the semantics of comparing floating point values.

Reply:
We could. I put this in the RowContainer because the test is at that level and it contains the simple type tests - so they are all together.
In the RowContainer there is a function that operates on the vector level (as the row must not be serialized) which is called `comparePrimitiveAsc`. Perhaps `native` isn't quite the right word for it and I should use `comparePrimitive`?

Reply:
Is this a copy of comparePrimitiveAsc from SimpleVector.h? If so, perhaps, modify SimpleVector.h to make comparePrimitiveAsc a public static method and then re-use here.

Reply:
No. It is a private static function in `RowContainer` (see RowContainer.h). This function also always compares in ascending order (and the caller takes care of switching the result based on the compare flag) while the new function allows for also descending order results by passing the compare flags. But otherwise, they are the same. I can see that there should be some synergy here.
Interesting, the RowContainer version is a copy of the SimpleVector one.

Reply:
It would be nice to refactor to ensure there is only one copy of this logic.

Reply:
I've refactored the code to use the SimpleVector implementation.

As for the comments of the ContainerRowSerde API. Isn't this more generic around data type value handling for floating point, similar to how Spark has it documented?
Maybe something for develop/types.rst?

Reply:
> I've refactored the code to use the SimpleVector implementation.

Thanks.

> As for the comments of the ContainerRowSerde API. Isn't this more generic around data type value handling for floating point, similar to how Spark has it documented?
> Maybe something for develop/types.rst?

It would be nice to have documentation both in the code and on the website.